src/data_info/services/entry.clj

```diff
@@ -188,37 +191,30 @@
 (defn- paged-dir-listing
   "Provides paged directory listing as an alternative to (list-dir). Always contains files."
   [irods user path entity-type bad-indicator sfield sord offset limit info-types]
-  (let [;; first, icat stuff in futures
-        zone         (cfg/irods-zone)
-        total        (future (icat/number-of-items-in-folder user zone path entity-type info-types))
-        total-bad    (future (total-bad user zone path entity-type info-types bad-indicator))
-        page         (future (icat/paged-folder-listing
-                       :user           user
-                       :zone           zone
-                       :folder-path    path
-                       :entity-type    entity-type
-                       :info-types     info-types
-                       :sort-column    sfield
-                       :sort-direction sord
-                       :limit          limit
-                       :offset         offset))
-        ;; now the irods stuff to run parallel to the icat stuff
-        id           (irods/lookup-uuid irods path)
+  (let [zone         (cfg/irods-zone)
+        page         (rods/folder-listing irods user zone path
+                                          :entity-type entity-type
+                                          :info-types info-types
+                                          :sort-column sfield
+                                          :sort-direction sord
+                                          :limit limit
+                                          :offset offset)
+        total        (delay (force page) @(rods/items-in-folder irods user zone path :entity-type entity-type :info-types info-types))
+        id           (rods/uuid irods user zone path)
         bad?         (is-bad? bad-indicator path)
-        perm         (perm/permission-for irods user path)
-        stat         (item/stat irods path)
-        date-created (:date-created stat)
-        mod-date     (:date-modified stat)
+        perm         (rods/permission irods user zone path)
+        date-created (rods/date-created irods user zone path)
+        mod-date     (rods/date-modified irods user zone path)
         name         (fs/base-name path)]
-    (clean-return irods ;; ensure that the with-jargon is closed out
-      (merge (fmt-entry id date-created mod-date bad? nil path name perm 0)
+    (clean-return (:jargon irods) ;; ensure that the with-jargon is closed out
+      (merge (fmt-entry @id @date-created @mod-date bad? nil path name @perm 0)
              (page->map (partial is-bad? bad-indicator) @page)
              {:total    @total
-              :totalBad @total-bad}))))
+              :totalBad 0}))))
```

Comment:
I'm counting "removing totalBad" as somewhat out of scope. It was returning 0 every time anyway, and nothing actually uses it, so this'll do for now.